src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -148,6 +148,52 @@ def rename_fs_volume(self, volname, newvolname, sure):
 
         return rename_volume(self.mgr, volname, newvolname)
 
+    def volume_info(self, **kwargs):
+        ONE_GB = 10 ** 9
+        ret     = None
+        volname = kwargs['vol_name']
+
+        try:
+            with open_volume(self, volname) as fs_handle:
+                path = self.volspec.base_dir
+                vol_info_dict = {}
+                try:
+                    st = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_SIZE,
+                                         cephfs.AT_SYMLINK_NOFOLLOW)
+
+                    usedbytes = st['size']
+                    vol_info_dict = get_pending_subvol_deletions_count(path)
+                    vol_info_dict['size'] = f"{int(usedbytes)} B"
+                except cephfs.Error as e:
+                    if e.args[0] == errno.ENOENT:
+                        pass
+                df = self.mgr.get("df")
+                pool_stats = dict([(p['id'], p['stats']) for p in df['pools']])
+                osdmap = self.mgr.get("osd_map")
+                pools = dict([(p['pool'], p) for p in osdmap['pools']])
+                metadata_pool_id, data_pool_ids = get_pool_ids(self.mgr, volname)
+                vol_info_dict["pools"] = {"metadata": [], "data": []}
+                for pool_id in [metadata_pool_id] + data_pool_ids:
+                    if pool_id == metadata_pool_id:
+                        pool_type = "metadata"
+                    else:
+                        pool_type = "data"
+                    vol_info_dict["pools"][pool_type].append({
+                                    'name': pools[pool_id]['pool_name'],
+                                    'used': f"{(pool_stats[pool_id]['bytes_used'] / ONE_GB):.9f} GB",
```

Comment:
I suggest leaving 'used' and 'avail' pool size in bytes. Having size in GB up to nine decimal places doesn't seem friendly to humans/applications who/that would consume this output. 

I was looking at `fs status` command output, which gives the exact same info, used and available sizes of data and metadata pools.

For applications, it provides "used" and "avail" pool sizes in bytes when `ceph fs status --format=json` is executed.
See,
```
 ./bin/ceph fs status --format=json-pretty
{
    "clients": [
        {
            "clients": 0,
            "fs": "a"
        }
    ],
    "mds_version": [
        {
            "daemon": [
                "a"
            ],
            "version": "ceph version 17.0.0-13762-ga421cfc2aa0 (a421cfc2aa0eef2d1ef171ee706c24aa0e8234bb) quincy (dev)"
        }
    ],
    "mdsmap": [
        {
            "caps": 0,
            "dirs": 16,
            "dns": 16,
            "inos": 19,
            "name": "a",
            "rank": 0,
            "rate": 0,
            "state": "active"
        }
    ],
    "pools": [
        {
            "avail": 106288873472,
            "id": 2,
            "name": "cephfs.a.meta",
            "type": "metadata",
            "used": 81920
        },
        {
            "avail": 106288873472,
            "id": 3,
            "name": "cephfs.a.data",
            "type": "data",
            "used": 4096
        }
    ]
}
```

Human readable pool sizes are shown when `ceph fs status` is executed.
```
[rraja@localhost build]$ ./bin/ceph fs status
a - 0 clients
=
RANK  STATE   MDS     ACTIVITY     DNS    INOS   DIRS   CAPS  
 0    active   a   Reqs:    0 /s    16     19     16      0   
     POOL        TYPE     USED  AVAIL  
cephfs.a.meta  metadata  80.0k  98.9G  
cephfs.a.data    data    4096   98.9G  
MDS version: ceph version 17.0.0-13762-ga421cfc2aa0 (a421cfc2aa0eef2d1ef171ee706c24aa0e8234bb) quincy (dev)
```

In this PR,  let's leave the pool sizes in bytes. We can later implement a "--human-readable" or "-h" flag that shows the used/avail pool sizes in KB/MB/GB by using mgr_utils.format_bytes() .

See,  
https://github.com/ceph/ceph/commit/4da6381fb1ec#diff-dc6485f717f4dce4863733896375af75963412ebb2abc4b62fcd1f5233eee07dR185
and
https://github.com/ceph/ceph/commit/4da6381fb1ec#diff-dc6485f717f4dce4863733896375af75963412ebb2abc4b62fcd1f5233eee07dR196

Reply:
> In this PR, let's leave the pool sizes in bytes. We can later implement a "--human-readable" or "-h" flag that shows the used/avail pool sizes in KB/MB/GB by using mgr_utils.format_bytes() .

OK -- that makes sense. It was me who brought this point up initially. @neesingh-rh pls do the required changes and a follow up PR implementing -h/--human-readable option.

Reply:
got it